### DISCUSSION TITLE
HTML element id

### DISCUSSION THREAD

**Discussion by marco-santulli** [Budibase/budibase#3910] [General]
Hi,

I notice there are often (Buttons, links in the top menu) no ids on html elements of BB.   This makes it not usable in case of RPA or other screen-scraping scenarios, can we add id's somehow?

M

**Comment by mjashanks** [comment]
Although we don't set the 'id' tag, we do give components an ID via 'class'

![image](https://user-images.githubusercontent.com/3524181/148050977-d62e33a2-9cd2-49d9-8051-ba703e0e5673.png)

i.e. `class="component <id> ..."`

Does that help?

Out of interest, why do you need to screen scrape? Would it be easier if you had access to the budibase directly?

**Reply by marco-santulli** [reply]
Hi Michael,

let me try to clarify: lots of companies are starting to use LCAP along RPA platforms.  In one instance where I am trying to use budi, we also use a Robotic Process Automation platform.  These platforms rely heavily on good and stable ID's for robustness of the robot.  So if the ID is "artificial" it should not change across builds as much sas possibile. 

I have the feeling adding ID's should be relatively simple (classic 20% effort for 80% of the result is good), and will pay dividends as budi takes hold in the enterprise market

M

**Reply by mjashanks** [reply]
Yea, I think it should be easy but not sure. And the existing component id's do not change unless you delete and re-add them. Will raise as an issue.

**Reply by marco-santulli** [reply]
excellent! thank you very much

M
